builtin/providers/azurerm/resource_arm_virtual_machine.go

```diff
@@ -190,7 +208,24 @@ func resourceArmVirtualMachine() *schema.Resource {
 
 						"vhd_uri": {
 							Type:     schema.TypeString,
-							Required: true,
+							Optional: true,
```

Comment:
I've been having an issue with using `ConflictsWith` since these are Set/List objects in the schema. I can get it to work with `storage_os_disk` since I can assume there is only a single entry and have the `ConflictsWith` set to `storage_os_disk.0.managed_disk_type` but I cannot do that with data disks since there could be n number of entries there and just specifying `storage_data_disk.managed_disk_type` does not work. Am I missing something here that could get this to work? If not, I can go about trying to implement proper support for list/set objects.